packages/rewards-calculator/src/workers.ts

```diff
@@ -49,18 +50,31 @@ export class Workers {
 
   constructor(private clickhouseClient: ClickhouseClient) {}
 
-  public add(workerId: string) {
-    if (this.workers[workerId]) {
-      return this.workers[workerId];
+  public add(peerId: string) {
+    if (this.workers[peerId]) {
+      return this.workers[peerId];
     }
-    this.workers[workerId] = new Worker(workerId);
-    return this.workers[workerId];
+    this.workers[peerId] = new Worker(peerId);
+    return this.workers[peerId];
   }
 
   public map<T>(fn: (worker: Worker, index: number) => T) {
     return Object.values(this.workers).map(fn);
   }
 
+  filterChunk(chunkType: 'odd' | 'even') {
+    const newWorkers =  Object.values(this.workers).filter(w => {
+      const arr = bs58.decode(w.peerId)
+      const isEven = arr[arr.length - 1] % 2 === 0;
+
+      return chunkType === 'even' ? isEven : !isEven;
+    });
+
+    this.workers = Object.fromEntries(newWorkers.map(w => [w.peerId, w]));
```

Comment:
Do we have to mutate `this` instead of just returning a new list as in the `map` function?

Reply:
Worker id from the contract will be reset and we have a ref cycle in Clickhouse client.
It is just safer in this design, which is not the best, to 


To make it works with new instance, it will look like 
```ts
filterChunk(chunkType: 'odd' | 'even') {
    const newWorkers =  Object.values(this.workers).filter(w => {
      const arr = bs58.decode(w.peerId)
      const isEven = arr[arr.length - 1] % 2 === 0;

      return chunkType === 'even' ? isEven : !isEven;
    });

    const newWorkersMap = new Workers(this.clickhouseClient, Object.fromEntries(newWorkers.map(w => [w.peerId, w]));
    
    this.clickhouseClient.workers = newWorkersMap;
    return newWorkersMap
  }
  ```

Other methods like `map` already mutate this.workers, 
not by changing the length, but by mutating it.